website/src/_posts/2022-09-3.0.md

```diff
@@ -0,0 +1,126 @@
+
+## Devotedly stable
+
+Since the Uppy 2.7.0 release, **we’ve fixed over 95 bugs!** This includes bugs that made uploading less stable, documentation corrections, and dependency upgrades.
+
+## Companion turns 4.0
+
+Everyone’s favorite Companion also received some love.
+
+### Streaming upload
+
+Streaming upload can now also be enabled in Companion when using Tus. This comes with greatly improved upload speeds and allows uploading potentially hundreds of gigabytes without needing a large server storage. This feature was also available before Companion 4, but it didn’t work with Tus yet.
+
+With streaming upload disabled (default), the whole file will be downloaded first, and only when the download has completely finished would the upload start.
+
+When streaming upload is enabled, Companion will start downloading the file from the provider (for example Google Drive), while at the same time starting the upload to the destination (for example Tus), and send every chunk of data consecutively.
+
+For more information, [see the docs](https://uppy.io/docs/companion/).
+
+### Event emitter
+
+Companion now has a server-side event emitter that can be used to detect when uploads start, finish and fail, without having to depend on the client (so it works even if the user has closed their browser).
+
+Example code:
+
+```js
+const { app, emitter } = companion.app(options)
+
+emitter.on('upload-start', ({ token }) => {
+  console.log('Upload started', token)
+
+  function onUploadEvent ({ action, payload }) {
+    if (action === 'success') {
+      emitter.off(token, onUploadEvent) // avoid listener leak
+      console.log('Upload finished', token, payload.url)
+    } else if (action === 'error') {
+      emitter.off(token, onUploadEvent) // avoid listener leak
+      console.error('Upload failed', payload)
+    }
+  }
+  emitter.on(token, onUploadEvent)
+})
+```
+
+### Internal improvements
+
+Companion’s internal request code and Providers have been rewritten to `async`/`await` which has grossly simplified and reduced our amount of code. This will lead to more less bugs and security issues. It will also make it easier to implement new custom Providers.
+
+We have also replaced the deprecated [request](https://github.com/request/request) library with [got](https://github.com/sindresorhus/got). This also removed the need for [purest](https://github.com/simov/purest).
+
+### And more!
+
+* [Compressor](/docs/compressor) is officially stable. The Compressor plugin for Uppy optimizes images (JPEG, PNG), saving on average up to 60% in size (roughly 18 MB for 10 images). It’s been battle tested and has proved itself.
```

Comment:
> roughly 18 MB for 10 images

Images are very different in terms of file size, so I'd propose to make the estimate more specific (e.g: 18 MB of random 10 iPhone JPG photos), or just remove the estimate at all — "Up to 60% in size" is already easy to understand.